### ISSUE TITLE
Some tests fail with ifort-2021.10 and ifx-2023.2

### ISSUE THREAD

**Issue by milancurcic** [nedtaylor/athena#28]
Based off the main branch, the library now builds fine with both ifort and ifx (thanks!). Running the tests, however, some fail. There are more failures with ifort than with ifx. This is built with fpm and the debug profile which enables various diagnostics. Running in the release mode simply turns these into segfaults. So I attach the outputs of

```
FPM_FC=ifort fpm test >& athena-fpm-test-log-ifort-2021.10.txt
FPM_FC=ifx fpm test >& athena-fpm-test-log-ifx-2023.2.txt
```

My full compiler versions are:

```
ifort --version
ifort (IFORT) 2021.10.0 20230609
Copyright (C) 1985-2023 Intel Corporation.  All rights reserved.

ifx --version
ifx (IFX) 2023.2.0 20230721
Copyright (C) 1985-2023 Intel Corporation. All rights reserved.
``` 

[athena-fpm-test-log-ifx-2023.2.txt](https://github.com/nedtaylor/athena/files/15472551/athena-fpm-test-log-ifx-2023.2.txt)
[athena-fpm-test-log-ifort-2021.10.txt](https://github.com/nedtaylor/athena/files/15472553/athena-fpm-test-log-ifort-2021.10.txt)

In support of https://github.com/openjournals/joss-reviews/issues/6492

**Comment by milancurcic** [comment]
FWIW, all tests seem to pass with gfortran-12 in both debug and release modes.

**Comment by nedtaylor** [comment]
Resolved in commit a8f0d619b0cdef6f3b1064367b400f7b16ab6772

**Comment by nedtaylor** [comment]
Merged into main
